tests/node-18.0/tests.js

```diff
@@ -1,37 +1,71 @@
-    
-    If an error is thrown, a response with code 500 will be returned.
-*/
+module.exports = async (context) => {
+    const action = context.req.headers['x-action'];
 
-module.exports = async (req, res) => {
-    const payload = JSON.parse(req.payload === '' ? '{}' : req.payload);
+    switch (action) {
+        case 'plaintextResponse':
+            return context.res.send('Hello World 👋');
+        case 'jsonResponse':
+            return context.res.json({ json: true, message: 'Developers are awesome.' });
+        case 'fileResponse':
+            return context.res.file(__dirname + '/resources/picture.png', undefined, { 'content-type': 'image/png' });
+        case 'redirectResponse':
+            return context.res.redirect('https://github.com/');
+        case 'emptyResponse':
+            return context.res.empty();
+        case 'noResponse':
+            context.res.send('This should be ignored, as it is not returned.');
+            break;
+        case 'doubleResponse':
+            context.res.send('This should be ignored.');
+            return context.res.send('This should be returned.');
+        case 'headersResponse':
+            return context.res.send('OK', 200, { 'first-header': 'first-value', 'second-header': 'second-value' });
+        case 'statusResponse':
+            return context.res.send('FAIL', 404);
+        case 'complexResponse':
+            context.res.setStatusCode(201);
+            context.res.setHeaders({ 'header1': 'value1' })
+            context.res.addHeader('header2', 'value2');
+            context.res.setBody('👌');
+            context.res.setContentType('application/json');
+            context.res.addCookie('cookie1', 'value1');
+            context.res.addCookie('cookie2', 'value2', 'Wed, 21 Oct 2015 07:28:00 GMT', 2592000, '/', 'google.com', true, true, 'Lax');
 
-    const todo = await fetch(`https://jsonplaceholder.typicode.com/todos/${payload.id ?? 1}`).then(r => r.json());
+            context.res.setBody(JSON.stringify({
+                code: context.res.getStatusCode(),
+                contentType: context.res.getContentType(),
+                body: context.res.getBody(),
+                headers: context.res.getHeaders(),
+                header1: context.res.getHeader('header1'),
+                header2: context.res.getHeader('header2'),
+                cookies: context.res.getCookies()
+            }));
+            return context.res.send();
+        case 'requestMethod':
+            return context.res.send(context.req.method);
+        case 'requestUrl':
+            return context.res.send(context.req.url);
+        case 'requestHeaders':
+            return context.res.json(context.req.headers);
+        case 'requestBodyPlaintext':
+            return context.res.send(context.req.body);
+        case 'requestBodyJson':
+            return context.res.send(context.req.body.data ?? 'Missing key');
```

Comment:
we also need tests for cases where the request is of type `multipart/form-data` or `application/x-www-form-urlencoded`

Reply:
Added support for form-data and url-encoded body.